### PR TITLE
[4.0] Add getMenu to the CMSApplicationInterface. Part of #24587

### PR DESCRIPTION
**Pull request by wilsonge**
Adds `getMenu` into the `\Joomla\CMS\Application\CMSApplicationInterface`. To test ensure menu's keep working in both the frontend and backend of the core

### CODE REVIEW

File: libraries/src/Application/CMSApplicationInterface.php [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 use Joomla\CMS\Extension\ExtensionManagerInterface;
 use Joomla\CMS\User\User;
+use Joomla\CMS\Menu\AbstractMenu;
```
Comment (Quy) [inline libraries/src/Application/CMSApplicationInterface.php:12]: Please sort in alpha order.

File: libraries/src/Application/CMSApplicationInterface.php [hunk 2]
```diff
@@ -154,6 +155,18 @@ public function isCli();
 	 */
 	public function getIdentity();
 
+	/**
+	 * Get the menu object.
+	 *
+	 * @param   string  $name     The application name for the menu
+	 * @param   array   $options  An array of options to initialise the menu with
+	 *
+	 * @return  AbstractMenu|null  A User object or null if not set.
```
Comment (SharkyKZ) [inline libraries/src/Application/CMSApplicationInterface.php]: Should be `AbstractMenu object`?

Comment (wilsonge) [reply libraries/src/Application/CMSApplicationInterface.php]: Oops. Good spot!

**Comment by SharkyKZ** [comment]
I have tested this item :white_check_mark: successfully on b9653f2a1ad3f89c3972c2869b954b59d7288d51<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24803">issues.joomla.org/tracker/joomla-cms/24803</a>.</sub>

**Comment by SniperSister** [comment]
@wilsonge ConsoleApplication is broken because of this merge:
> Class Joomla\CMS\Application\ConsoleApplication contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (Joomla\CMS\Application\CMSApplicationInterface::getMenu)

Making the ConsoleApplication implement the getMenu method feels weird, I would keep it in SiteApplication.

**Comment by wilsonge** [comment]
I don't think it should that way. Plugin's use it and plugins work across all application types including console. We shouldn't have to do method exists in every application. I think we probably need to do the same ugly thing as we do in the `InstallationApplication` and just have it `return null` consistently

**Comment by SniperSister** [comment]
Ok, I trust you here :) can you do the PR?

**Comment by wilsonge** [comment]
Thanks for doing it! Sorry been on a train all day today so no good interweb

**Pull request merged**